ts/Dashboards/EditMode/Sidebar.ts

```diff
@@ -831,17 +831,45 @@ class Sidebar {
             const activeTabContainer = activeTab && activeTab.content &&
                 activeTab && activeTab.content.container;
             let type;
+            let chartTypes = {};
 
             for (const key in componentSettings) {
                 if (componentSettings[key]) {
                     type = componentSettings[key].type;
 
+                    if (key === 'chartType') {
+                        const chartTypesEnum: Array<string> = [
+                            'column',
+                            'line',
+                            'scatter',
+                            'spline',
+                            'pie'
+                        ];
+
+                        const chartOpts =
+                            (currentComponent as any).chartOptions;
```

Comment:
Can we check if the `currentComponent` is a Highcahts Component that has `chartOptions`